### PR TITLE
Update `dropdown` doc string

### PR DESCRIPTION
**Pull request by KDecay**
I updated the `dropdown` doc string with the example from the [book](https://github.com/vizia/vizia-book/blob/develop/src/views/dropdown.md).

### CODE REVIEW

File: core/src/views/dropdown.rs [hunk 1]
```diff
@@ -3,11 +3,67 @@ use crate::{
     View,
 };
 
+/// The dropdown icon of a dropdown view.
 pub const ICON_DOWN_OPEN: &str = "\u{e75c}";
 
-pub struct Dropdown {}
+/// A dropdown used to display some state with the ability to open a popup with options to change that state.
```
Comment (geom3trik) [inline core/src/views/dropdown.rs:9]: Missing 'is'. "A dropdown **is** used to display..."

File: core/src/views/dropdown.rs [hunk 2]
```diff
@@ -3,11 +3,67 @@ use crate::{
...
+///
+/// Usually a dropdown is used in the context of a "combo box" or "list picker" to allow the user to select
```
Comment (geom3trik) [inline core/src/views/dropdown.rs:11]: I could be wrong but I always thought a combo box was specifically a dropdown + textbox combination.

File: core/src/views/dropdown.rs [hunk 3]
```diff
@@ -3,11 +3,67 @@ use crate::{
...
+/// #
+/// Dropdown::new(
+///     cx,
+///     |cx| Label::new(cx, AppData::value),
+///     |cx| {
+///         for i in 0..5 {
+///             Label::new(cx, i)
+///                 .on_press(move |cx| {
+///                     cx.emit(AppEvent::SetValue(i));
+///                     cx.emit(PopupEvent::Close); // close the popup
```
Comment (geom3trik) [inline core/src/views/dropdown.rs:45]: Could be worth mentioning here that closing the popup when selecting is optional. This allows for something like a dropdown where multiple things can be selected.

**Review by geom3trik — CHANGES_REQUESTED** [review]
I think we should add some more detail about what can be done with the dropdown. 
The first closure doesn't have to show the current option, it could just display some text, or an icon, or text + icon, or even a textbox.
The second closure is for the contents but it doesn't necessarily have to be a list or even a collection of the same views. For example, if you wanted a dropdown with multiple selectable items where each one has a checkbox and a label then you could build that.
The dropdown then is the bare bones of a view which has a "closed" view and an "open" view.

File: core/src/views/dropdown.rs (hunk 2 above)
Comment (rhelmot) [reply core/src/views/dropdown.rs:11]: I'm pretty sure you could use the Dropdown widget to implement that kind of form element as well!

**Review by rhelmot — APPROVED** [review]

**Review by geom3trik — APPROVED** [review]

**Pull request merged**